Add typescript

MelWttr opened the pull request.

Commit:
build: добавляет typescript в проект

Commit:
feature(index.ts): добавляет типизацию в код

taraswww777 commented on package.json:
```diff
@@ -15,6 +15,8 @@
   "author": "melWttr",
   "license": "ISC",
   "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^8.12.2",
```
### Нет `package-lock.json`

Нет файла `package-lock.json`, его надо добавить.

Чтобы можно при инициализации быстро установить все нужные зависимости командой `npm ci`.

taraswww777 commented on package.json:
```diff
@@ -15,6 +15,8 @@
   "license": "ISC",
   "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^8.12.2",
+    "@typescript-eslint/parser": "^8.12.2",
```
### Версии зависимостей в package.json

Лучше указывать точные версии зависимостей без символов `~` и `^`.
Например так: `"lodash":"4.17.21"`
Команда для установки `npm i lodash -ED`

taraswww777 commented on src/index.ts:
```diff
@@ -0,0 +1,83 @@
+const volume: HTMLInputElement = document.querySelector('.volume-bar input') as HTMLInputElement;
+const buttonsList: HTMLElement = document.querySelector('.weather-list') as HTMLElement;
+
+interface Weather {
```
```ts
interface WeatherItem {
  audio: HTMLAudioElement;
  bg: string;
}

type Weather = Record<string, WeatherItem>;

```

Commit:
fix: удаляет Index.js

Commit:
build: рефакторинг зависимостей

- возвращает package-lock из gitignore
- фиксирует зависисмости на конкретные версии

Commit:
fix(index.ts): исправляет баги button и volume

- добавляет возможность клика по свг - иконке
- добавляет проверку на null в обработчик слайдера

taraswww777 commented on src/index.ts:
```diff
@@ -0,0 +1,86 @@
+import './css/index.scss';
```
Этот файл должен быть в гитигнор

Commit:
Merge branch 'development' into add-typescript
